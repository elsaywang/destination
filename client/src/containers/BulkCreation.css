:local(.bulkCreation) {
    display: flex;
    justify-content: flex-end;
}
:local(.message) {
    display: flex;
    padding: 0.5em;
    font-weight: 800;
}
