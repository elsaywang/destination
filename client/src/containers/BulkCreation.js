import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Add from '@react/react-spectrum/Icon/Add';
import Button from '@react/react-spectrum/Button';
import styles from './BulkCreation.css';
import { connect } from 'react-redux';
import { createTraitFromMultiSignals } from '../actions';

export class BulkCreation extends Component {
    render() {
        const { createTraitFromMultiSignals, selectedSignals } = this.props;

        const { selectionMessage, records, warning } = selectedSignals;

        return records.length ? (
            <div className={styles.bulkCreation}>
                <span className={styles.message}>{selectionMessage}</span>
                <Button
                    label="Create Trait From Multi Signals"
                    icon={<Add />}
                    onClick={createTraitFromMultiSignals}
                    variant="action"
                    disabled={warning}
                />
            </div>
        ) : null;
    }
}

BulkCreation.propTypes = {
    selectedSignals: PropTypes.object,
    createTraitFromMultiSignals: PropTypes.func,
};

const mapStateToProps = ({ selectedSignals }) => ({
    selectedSignals,
});

export default connect(mapStateToProps, createTraitFromMultiSignals)(BulkCreation);
