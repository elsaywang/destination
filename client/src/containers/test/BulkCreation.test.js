import React from 'react';
import { shallow } from 'enzyme';
import { BulkCreation } from '../BulkCreation';
import Button from '@react/react-spectrum/Button';
import Add from '@react/react-spectrum/Icon/Add';

describe('<BulkCreation/> component', () => {
    describe('rendering when there is no signal selected', () => {
        const mockFn = jest.fn();
        const initialprops = {
            selectedSignals: {
                displayContext: '',
                createTraitFromMultiSignals: mockFn,
                records: [],
                warning: '',
            },
        };
        const wrapper = shallow(<BulkCreation {...initialprops} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders null element', () => {
            expect(wrapper.getElement()).toBe(null);
        });
    });

    describe('rendering when there are selectedSignals.records pass in props ', () => {
        const props = {
            selectedSignals: {
                displayContext: '1 Real-time signal selected ',
                records: [{ rowIndex: 0, signalType: 'Adobe Analytics' }],
                warning: '',
            },
        };
        const wrapper = shallow(<BulkCreation {...props} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders the displayContext passed from selectedSignals prop into a <span/>', () => {
            expect(
                wrapper.containsMatchingElement(
                    <span>{props.selectedSignals.displayContext}</span>,
                ),
            ).toBeTruthy();
        });

        it('renders <Button /> component with correct "Create Trait From Multi Signals" labeling', () => {
            expect(wrapper.find(Button).exists()).toBe(true);
            expect(wrapper.find(Button).props().label).toEqual('Create Trait From Multi Signals');
        });

        it('renders <Button /> component includes <Add/> as an icon prop', () => {
            expect(wrapper.find(Button).props().icon).toEqual(<Add />);
        });

        it('renders non-disabled <Button /> with "action" variant when no warning message passes from props', () => {
            expect(wrapper.find(Button).props().variant).toEqual('action');
            expect('disabled' in wrapper.props()).toBeFalsy();
        });

        it('renders disabled <Button /> with "action" variant when warning message passes from props', () => {
            const propsWithWarning = {
                selectedSignals: {
                    displayContext: '1 Real-time signal , 1 Onboarded signal selected ',
                    records: [
                        { rowIndex: 0, signalType: 'Adobe Analytics' },
                        { rowIndex: 1, signalType: 'Onboarded Records' },
                    ],
                    warning: 'There is error in your selected signals ! ',
                },
            };
            wrapper.setProps({ ...propsWithWarning });
            expect(wrapper.find(Button).props().variant).toEqual('action');
            expect(wrapper.find(Button).props().disabled).toBe(true);
        });
    });
});
