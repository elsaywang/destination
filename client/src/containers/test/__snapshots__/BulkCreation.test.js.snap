// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BulkCreation/> component rendering when there are selectedSignals.records pass in props  matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BulkCreation
    selectedSignals={
      Object {
        "records": Array [
          Object {
            "rowIndex": 0,
            "signalType": "Adobe Analytics",
          },
        ],
        "selectionMessage": "1 Real-time signal selected ",
        "warning": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className={undefined}
        >
          1 Real-time signal selected 
        </span>,
        <Button
          block={false}
          disabled={false}
          element="button"
          icon={<Add />}
          invalid={false}
          label="Create Trait From Multi Signals"
          logic={false}
          onClick={undefined}
          quiet={false}
          selected={false}
          variant="action"
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1 Real-time signal selected ",
          "className": undefined,
        },
        "ref": null,
        "rendered": "1 Real-time signal selected ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "disabled": false,
          "element": "button",
          "icon": <Add />,
          "invalid": false,
          "label": "Create Trait From Multi Signals",
          "logic": false,
          "onClick": undefined,
          "quiet": false,
          "selected": false,
          "variant": "action",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className={undefined}
          >
            1 Real-time signal selected 
          </span>,
          <Button
            block={false}
            disabled={false}
            element="button"
            icon={<Add />}
            invalid={false}
            label="Create Trait From Multi Signals"
            logic={false}
            onClick={undefined}
            quiet={false}
            selected={false}
            variant="action"
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1 Real-time signal selected ",
            "className": undefined,
          },
          "ref": null,
          "rendered": "1 Real-time signal selected ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "disabled": false,
            "element": "button",
            "icon": <Add />,
            "invalid": false,
            "label": "Create Trait From Multi Signals",
            "logic": false,
            "onClick": undefined,
            "quiet": false,
            "selected": false,
            "variant": "action",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<BulkCreation/> component rendering when there is no signal selected matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BulkCreation
    selectedSignals={
      Object {
        "createTraitFromMultiSignals": [MockFunction],
        "records": Array [],
        "selectionMessage": "",
        "warning": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
